src/shared/cpu-set-util.c

```diff
@@ -371,6 +372,45 @@ int apply_numa_policy(const NUMAPolicy *policy) {
         return 0;
 }
 
+int numa_to_cpu_set(const NUMAPolicy *policy, CPUSet *set) {
+        int r;
+        size_t i;
+        _cleanup_(cpu_set_reset) CPUSet s = {};
+
+        assert(policy);
+        assert(set);
+
+        for (i = 0; (size_t) i < policy->nodes.allocated * 8; i++) {
+                _cleanup_free_ char *p = NULL, *l = NULL;
+                _cleanup_(cpu_set_reset) CPUSet part = {};
+
+                if (!CPU_ISSET_S(i, policy->nodes.allocated, policy->nodes.set))
+                        continue;
+
+                r = asprintf(&p, "/sys/devices/system/node/node%lu/cpulist", (unsigned long) i);
+                if (r < 0)
+                        return -ENOMEM;
+
+                r = read_one_line_file(p, &l);
+                if (r < 0)
+                        return r;
+
+                r = parse_cpu_set(l, &part);
+                if (r < 0)
+                        return r;
+
+                r = cpu_set_add_all(&s, &part);
+                if (r < 0)
+                        return r;
+        }
+
+        r = cpu_set_add_all(set, &s);
+        if (r < 0)
+                return r;
+
+        return 0;
```

Comment:
so do i get this right, the conversion is different depending on the local system? if so, this is one reason more to do the conversion on demand rather than on load

Reply:
Hmm...Conversion logic is the same everywhere, of course, result may be different based on the local system.

Reply:
well, but the parsing should be the same everywhere. the application of the settings should deal with the differences, not the parsing. If I do systemctl show on such a device it should look the same everywhere really if i have the same config

Reply:
I am not sure I follow so let's use an example. Assuming I have following `test.service`,

```
[Service]
ExecStart=/bin/sleep 42
NUMAMask=1
NUMAPolicy=bind
CPUAffinity=numa
```

And I have two systems, A has 2 CPU cores per NODE node while B has 4 cores per each NUMA node. I'd think that loading above unit on both machines and then running `systemctl show -p CPUAffinity test.service` *should* produce different output and that is actually expected.

You think it shouldn't be the case?